jetty-core/jetty-start/src/main/java/org/eclipse/jetty/start/Main.java

```diff
@@ -154,8 +160,93 @@ private void dumpClasspathWithVersions(String name, PrintStream out, Classpath c
         int i = 0;
         for (Path element : classpath.getElements())
         {
-            out.printf("%2d: %24s | %s\n", i++, getVersion(element), baseHome.toShortForm(element));
+            String license = getLicenceFromJar(element);
+            if (license != null && !license.isEmpty())
+                out.printf("%2d: %24s | %s | %s\n", i++, getVersion(element), baseHome.toShortForm(element), license);
+            else
+                out.printf("%2d: %24s | %s\n", i++, getVersion(element), baseHome.toShortForm(element));
+        }
+    }
+
+    private String getLicenceFromJar(Path jar)
+    {
+        if (!Files.exists(jar) || Files.isDirectory(jar) || !Files.isReadable(jar))
+            return null;
+        try
+        {
+            try (JarFile jarFile = new JarFile(jar.toFile()))
+            {
+                Manifest manifest = jarFile.getManifest();
+                if (manifest != null)
+                {
+                    String spdxLicense = manifest.getMainAttributes().getValue("SPDX-License-Identifier");
+                    if (spdxLicense != null)
+                        return spdxLicense;
+
+                    String bundleLicense = manifest.getMainAttributes().getValue("Bundle-License");
+                    if (bundleLicense != null)
+                        return bundleLicense;
+                }
+
+                Optional<String> license = jarFile.stream().filter(Main::isLicenseFile).map(e -> getLicenceFromFile(jarFile, e)).filter(Objects::nonNull).findFirst();
+                if (license.isPresent())
+                    return license.get();
+
+            }
+        }
+        catch (Throwable ignored)
+        {
+        }
+        return null;
+    }
+
+    private static boolean isLicenseFile(JarEntry entry)
+    {
+        String name = entry.getName();
+        return name.matches("(?i)^(META-INF/)?LICENSE.*") || name.matches("(?i)^LICENSE.*");
```

Comment:
Need "license" and "licence" (both are found in the wild)

https://www.grammarly.com/blog/licence-license/

Reply:
Note: `/about.html` is also a place to look for licenses.
Take the above `ecj-3.37.0.jar` as an example.

Reply:
The `/META-INF/maven/<groupId>/<artifactId>/pom.xml` can contain the license in two different ways.

As a formal `<licenses>` element in the XML (might be in the parent pom, but that's too much work to fish for), and also as a comment in the `pom.xml` itself.

Take `animal-sniffer-annotations-1.23.jar` as an example of the latter (as an XML comment)

Reply:
I hear you about `about.html` and the pom elements.... but worried that might get a bit too complex.

Reply:
The nature of LICENSE is rather complex.
Listing just the URLs is, to me, a no-go, as it doesn't convey the LICENSE details properly.

Example:

* In the case of more than one license, is the choice "License-A AND License-B" or "License-A OR License-B" ?
* In the case of GPL and LGPL, is the text referencing the license mentioning "only" or "or later", this distinction is present in the SPDX ids, but not if you just use the URLs.
* And many of the other things that Apache JMeter brought up with the SPDX folks 4 years ago - https://lists.spdx.org/g/Spdx-tech/message/3866

Personally, the URL list in the output is a no-go.
We should be SPDX or nothing else.
That would also simplify this implementation, dropping the entire SPDX url list class this PR has.

Reply:
@joakime I don't understand what the problem with URIs is?   There is no semantic difference really between `Apache-2.0` and `https://www.apache.org/licenses/LICENSE-2.0.txt`, except the later is a little bit more convenient if you are looking.
The issue with multiple jars is not about how they are referenced, but how they are combined.   The bundle plugin has chosen to us a comma rather than "or" in the bundle-license field... but is that really a problem?  Are there any examples of AND for multiple licenses? If so, what does that even mean?

Anyway, for our jars, I have added to this PR that we set the SPDX to `EPL-2.0 OR Apache-2.0` in the manifest of all our jars, so that overrides the comma confusion of the bundle-license (which is still there).

But if a jar does have an SPDX, but does have a bundle-license with commas, then I see no reason not to list that, even as a URL, because that is what that jar is telling us the license is.  Others will need to look closer if they wish to interpret.  Hiding information doesn't change that.

Ditto for URIs found in LICENSE files.  They may be wrong or miss directions, but this list is only a convenience and nothing legal.  So better to tell the info we have found and if it is not clear then others can look deeper.